File: resolve_home.py

import os

file_path = 'pages/Home.tsx'

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

new_lines = []
in_head = False
in_incoming = False

for line in lines:
    if line.startswith('<<<<<<< HEAD'):
        in_head = True
        continue
    elif line.startswith('======='):
        in_head = False
        in_incoming = True
        continue
    elif line.startswith('>>>>>>>'):
        in_incoming = False
        continue
    
    if in_head or (not in_head and not in_incoming):
        new_lines.append(line)

with open(file_path, 'w', encoding='utf-8') as f:
    f.writelines(new_lines)

print("Resolved conflicts in Home.tsx")

File: resolve_home.js

const fs = require('fs');
const path = 'pages/Home.tsx';

try {
    const data = fs.readFileSync(path, 'utf8');
    const lines = data.split(/\r?\n/);
    const newLines = [];
    let state = 'normal'; // normal, head, incoming

    for (let line of lines) {
        if (line.startsWith('<<<<<<< HEAD')) {
            state = 'head';
            continue;
        } else if (line.startsWith('=======')) {
            state = 'incoming';
            continue;
        } else if (line.startsWith('>>>>>>>')) {
            state = 'normal';
            continue;
        }

        if (state === 'normal' || state === 'head') {
            newLines.push(line);
        }
    }

    fs.writeFileSync(path, newLines.join('\n'), 'utf8');
    console.log("Resolved conflicts in Home.tsx");
} catch (err) {
    console.error(err);
}
